scanner.go

```diff
@@ -359,3 +378,63 @@ func (s *scanner) closeRegionScanner() {
 	}
 	s.curRegionScannerID = noScannerID
 }
+
+// renews a scanner by resending scan request with renew = true
+func (s *scanner) renew() error {
+	if err := s.rpc.Context().Err(); err != nil {
+		return err
+	} else {
+		rpc, err := hrpc.NewScanRange(s.rpc.Context(),
+			s.rpc.Table(),
+			s.startRow,
+			nil,
+			hrpc.ScannerID(s.curRegionScannerID),
+			hrpc.NumberOfRows(s.rpc.NumberOfRows()),
+			hrpc.Priority(s.rpc.Priority()),
+			hrpc.RenewalScan(),
+		)
+		if err != nil {
+			return err
+		}
+		res, err := s.SendRPC(rpc)
+		if err != nil {
+			return err
+		}
+		resp, ok := res.(*pb.ScanResponse)
+		if !ok {
+			return errors.New("got non-ScanResponse for scan request")
+		}
+		// save new scanner id
+		s.curRegionScannerID = resp.GetScannerId()
+		return nil
+	}
+}
+
+// starts the renew timer, after it fires, renewScanner is called
+func (s *scanner) startRenewTimer() {
+	if s.renewTimer != nil {
+		panic("Renew timer was not stopped before starting again")
+	}
+	if s.rpc.RenewInterval() > 0 {
+		s.renewTimer = time.AfterFunc(s.rpc.RenewInterval(), s.renewScanner)
+	}
+}
+
+// stops the renewal timer
+func (s *scanner) stopRenewTimer() {
+	if s.renewTimer != nil {
+		s.renewTimer.Stop()
+		s.renewTimer = nil
+	}
+}
+
+// renews the scanner and restarts timer after renewal
+func (s *scanner) renewScanner() {
+	err := s.renew()
+	if err != nil {
+		s.logger.Debug("Error in renewing: ", slog.String("err: ", err.Error()))
```

Comment:
This log at the error level. And because this is a structured logger you don't need to include ": " in your strings. So more like:

`s.logger.Error("error renewing scanner", "err", err)`

Reply:
Fixed!